### PR TITLE
remove strong type on SendFeedbackForFeature function

### LINKED ISSUE #18537: Fatal error: In Feedback.sendFeedbackForFeature API
Getting below error.

> Error: {"message":"Argument 2 passed to Piwik\\Plugins\\Feedback\\API::sendFeedbackForFeature() must be of the type int, string given","file":"plugins\/Feedback\/API.php","line":38," "request_id":"d5fc2","backtrace":" on /plugins\/Feedback\/API.php(38)\n#0 [internal function]: Piwik\\Plugins\\Feedback\\API->sendFeedbackForFeature('Help', 'index.php\/.', 'dislikeother', 'e')\

I noticed this API is enforcing specific types. I'm thinking in API's we should likely not make use of PHP types like `(?string $featureName, int $like` as it could cause issues like fatal errors when you actually want a regular error message response that the input is not correct.

Could we maybe remove those types there and in general document to not use any types there? We could maybe even have an automated test that checks that no API has specific parameters enforced.

It can be frustrating otherwise when consuming the API and all you get is an HTTP 500 and you don't know what you're doing wrong.  Also we get emailed about it every time this happens which can be annoying too.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Fixes: #18537 
remove strong type on SendFeedbackForFeature function

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Feedback/API.php [hunk 1]
```diff
@@ -31,11 +31,11 @@ class API extends \Piwik\Plugin\API
      * config: "feedback_email_address".
      *
      * @param string|null $featureName  The name of a feature you want to give feedback to.
-     * @param int         $like         Whether you like the feature or not
+     * @param string|null $like         Whether you like the feature or not
```
Comment (justinvelluppillai) [inline plugins/Feedback/API.php:34]: I guess this is a bit strange to use `string|null` and then set the default to `$like = false`.

File: plugins/Feedback/API.php [hunk 2]
```diff
@@ -31,11 +31,11 @@ class API extends \Piwik\Plugin\API
...
      * @param string|null $choice       Multiple choice option chosen
      * @param string|null $message      A message containing the actual feedback
      */
-    public function sendFeedbackForFeature(?string $featureName, int $like, ?string $choice, ?string $message = null)
+    public function sendFeedbackForFeature($featureName, $like = false, $choice = null, $message = null)
```
Comment (justinvelluppillai) [inline plugins/Feedback/API.php]: Maybe this should be `$like = null`

File: plugins/Feedback/vue/src/RateFeature/RateFeature.vue [hunk 3]
```diff
@@ -257,7 +257,7 @@ export default defineComponent({
       AjaxHelper.fetch({
         method: 'Feedback.sendFeedbackForFeature',
         featureName: this.title,
-        like: this.like ? '1' : '0',
```
Comment (justinvelluppillai) [inline plugins/Feedback/vue/src/RateFeature/RateFeature.vue:260]: Suggest keeping this line as is otherwise both `'true'` and `'false'` may be true in php

**Review by justinvelluppillai — CHANGES_REQUESTED** [review]
Left a couple comments needing fixing

**Review by justinvelluppillai — APPROVED** [review]

**Comment by sgiehl** [comment]
@justinvelluppillai @peterhashair didn't you see that the vue build action failed for this PR due to some code formatting issues? Would be good to always check that before merging, as now the action fails for all PRs after merging in latest changes...
We maybe should consider to require this action to pass for a PR to be mergeable.

**Comment by justinvelluppillai** [comment]
@sgiehl sorry missed that - yes certainly the build should always be required to pass at least before merging.

**Comment by sgiehl** [comment]
I have pushed a fix here: https://github.com/matomo-org/matomo/pull/18525/
Will merge the PR soon, so it will be fixed on 4.x-dev as well.

**Comment by peterhashair** [comment]
@sgiehl yes, sorry, I forgot to update the VUE format.

**Pull request merged**